{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "collapsed_sections": []
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    }
  },
  "cells": [
    {
      "cell_type": "markdown",
      "source": [
        "**Подготовка файлов и ПО**\n"
      ],
      "metadata": {
        "id": "aiLriL4wj27P"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!pwd\n",
        "!rm -rf *"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "jq6a4S8VmiJi",
        "outputId": "a3f743a6-0723-457b-ca90-f786d26bc299"
      },
      "execution_count": 24,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "/content\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Файл из прошлого дз"
      ],
      "metadata": {
        "id": "muJRBiSykYqP"
      }
    },
    {
      "cell_type": "code",
      "execution_count": 39,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "BdMfT2oxjvKJ",
        "outputId": "84fffce2-bbfc-4799-d5db-7e715b08ffaa"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-21 13:38:21--  https://raw.githubusercontent.com/ZhnecShvec/hse22_hw1/main/platanus/Poil_gapClosed.fa\n",
            "Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 3962004 (3.8M) [text/plain]\n",
            "Saving to: ‘Poil_gapClosed.fa’\n",
            "\n",
            "Poil_gapClosed.fa   100%[===================>]   3.78M  --.-KB/s    in 0.07s   \n",
            "\n",
            "2022-10-21 13:38:21 (54.4 MB/s) - ‘Poil_gapClosed.fa’ saved [3962004/3962004]\n",
            "\n"
          ]
        }
      ],
      "source": [
        "!wget https://raw.githubusercontent.com/ZhnecShvec/hse22_hw1/main/platanus/Poil_gapClosed.fa"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!head /content/Poil_gapClosed.fa\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "1o8ljaMGwy2h",
        "outputId": "969c326d-0202-4277-fd03-59e1c95eade6"
      },
      "execution_count": 40,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            ">scaffold1_cov231\n",
            "TGAATTTTTAAAGAACGTTGCTTTGATTCTCTCAAAGCGGGCTGAGAATTATACATTAGCAGCTTTGTTTGTCAATCTAT\n",
            "TTTCTTAATCATTTTAATAAGGATTAAGCGAACAAACCAACCGCAAACTCAAACCAGAAGGCCTTCATTTGCAAGTGCAT\n",
            "CTTACCGCACCTTTAATCTATCCTGCAAGCTTTTTATTTTAAAATTTCGTTTAAAATCAAACACTTACAGAATCTTAGAG\n",
            "ATTCACCCGAAGGCGTCACCCTCTCAAAACAGGCTGCAAATCTTACGCTAATTTTGACTTTCGTCAATACCTTTCTTGAG\n",
            "AAACTTTTCGAAGACTTAAAAACCTAAAACAATCACTAAGTCATTGAAAAATATGGAGTTTTCACTCCGACTTAAGCGCC\n",
            "GCAGCGATCAAGAGGTGCGCATTATAGAGACATCAGTTCTGCCGTCAACAGGTTTATGAAAGTTTTTTTCAAAGGCTGCT\n",
            "GCAAAGTGCGTATACTTGCCACTCTTATATAGAAAGCACCGTTAACTCCCTACTGAGCTAGTGTTTAGTTCCAGTATTGC\n",
            "TATAGAGGTTCCCATGTCGGAGAAAGCTGTTGTCATTTATTCAGGTGGCATGGATTCGTTCACCATTCTTAATAAAGCCA\n",
            "AGCAAGAAGGTTATGAGCTGTATGCCCTCACCTTCGATTATGGTCAGAAGCATAAGAAAGAGATTCAGTACGCTATCTCT\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Все ПО:"
      ],
      "metadata": {
        "id": "ME2_t7iPkW7A"
      }
    },
    {
      "cell_type": "markdown",
      "source": [
        "Скачивание подходящей версии GeneMark для UNIX-64"
      ],
      "metadata": {
        "id": "tSH24PVClh_C"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://topaz.gatech.edu/GeneMark/tmp/GMtool_2qeB3/gms2_linux_64.tar.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "aRVfpXETlhCz",
        "outputId": "9d3de553-2799-463a-f9ec-f8adfbaa51c6"
      },
      "execution_count": 26,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-21 13:33:39--  http://topaz.gatech.edu/GeneMark/tmp/GMtool_2qeB3/gms2_linux_64.tar.gz\n",
            "Resolving topaz.gatech.edu (topaz.gatech.edu)... 143.215.161.100\n",
            "Connecting to topaz.gatech.edu (topaz.gatech.edu)|143.215.161.100|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 11261753 (11M) [application/x-gzip]\n",
            "Saving to: ‘gms2_linux_64.tar.gz’\n",
            "\n",
            "gms2_linux_64.tar.g 100%[===================>]  10.74M  9.87MB/s    in 1.1s    \n",
            "\n",
            "2022-10-21 13:33:40 (9.87 MB/s) - ‘gms2_linux_64.tar.gz’ saved [11261753/11261753]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Получение ключа"
      ],
      "metadata": {
        "id": "RqXJrp44mItO"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget http://topaz.gatech.edu/GeneMark/tmp/GMtool_2qeB3/gm_key_64.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "353vjKkAmJS0",
        "outputId": "7c59ce51-da1a-4428-8c0a-faf8558a772d"
      },
      "execution_count": 27,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-21 13:33:40--  http://topaz.gatech.edu/GeneMark/tmp/GMtool_2qeB3/gm_key_64.gz\n",
            "Resolving topaz.gatech.edu (topaz.gatech.edu)... 143.215.161.100\n",
            "Connecting to topaz.gatech.edu (topaz.gatech.edu)|143.215.161.100|:80... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 85 [application/x-gzip]\n",
            "Saving to: ‘gm_key_64.gz’\n",
            "\n",
            "\rgm_key_64.gz          0%[                    ]       0  --.-KB/s               \rgm_key_64.gz        100%[===================>]      85  --.-KB/s    in 0s      \n",
            "\n",
            "2022-10-21 13:33:40 (15.0 MB/s) - ‘gm_key_64.gz’ saved [85/85]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Установка и активация"
      ],
      "metadata": {
        "id": "lP0egxoDmSka"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!gzip -d gm_key_64.gz\n",
        "!tar -xzvf gms2_linux_64.tar.gz\n",
        "!cp  -v  gm_key_64   ~/.gmhmmp2_key"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "JKsZJCLKmTDj",
        "outputId": "f9836dce-3c9d-454a-8972-d90872b048fa"
      },
      "execution_count": 28,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "gms2_linux_64/\n",
            "gms2_linux_64/gms2.pl\n",
            "gms2_linux_64/README_GMS2.txt\n",
            "gms2_linux_64/mgm_4.mod\n",
            "gms2_linux_64/gmhmmp2\n",
            "gms2_linux_64/INSTALL\n",
            "gms2_linux_64/mgm_11.mod\n",
            "gms2_linux_64/biogem\n",
            "gms2_linux_64/LICENSE\n",
            "gms2_linux_64/mgm_25.mod\n",
            "gms2_linux_64/compp\n",
            "gms2_linux_64/mgm_15.mod\n",
            "'gm_key_64' -> '/root/.gmhmmp2_key'\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Установка E-Utilities "
      ],
      "metadata": {
        "id": "xzxSJAWbnUNC"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!sh -c \"$(curl -fsSL ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/install-edirect.sh)\"\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Twi32Y0YmTkr",
        "outputId": "8b4cea43-774e-42dc-90ab-bf19c8d40f6a"
      },
      "execution_count": 29,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "Entrez Direct has been successfully downloaded and installed.\n",
            "\n",
            "\n",
            "To activate EDirect for this terminal session, please execute the following:\n",
            "\n",
            "export PATH=${PATH}:${HOME}/edirect\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "!echo \"export PATH=\\${PATH}:/root/edirect\" >> ${HOME}/.bashrc"
      ],
      "metadata": {
        "id": "L0RjBNCWno30"
      },
      "execution_count": 30,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Файлы с данными бактерии T.oleivorans"
      ],
      "metadata": {
        "id": "0V88VQF_ntpR"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!$HOME/edirect/efetch -db nuccore -id HF680312 -format gb  >  T_oleivorans_MIL_1.gbk\n",
        "!$HOME/edirect/efetch -db nuccore -id HF680312 -format gene_fasta  >  T_oleivorans_MIL_1.genes.fasta\n",
        "!$HOME/edirect/efetch -db nuccore -id HF680312 -format fasta_cds_aa  >  T_oleivorans_MIL_1.proteins.fasta"
      ],
      "metadata": {
        "id": "gcwX2agkn2Id"
      },
      "execution_count": 31,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "Установка BLAST:"
      ],
      "metadata": {
        "id": "raqBEJABn4Yz"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!apt-get update\n",
        "!apt-get install ncbi-blast+"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "j1jeXU5Ip3xD",
        "outputId": "6e70e8b7-44a9-4d1d-e34d-c1117d2613eb"
      },
      "execution_count": 32,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rHit:1 http://security.ubuntu.com/ubuntu bionic-security InRelease\n",
            "\r0% [Connecting to archive.ubuntu.com (185.125.190.36)] [Connecting to cloud.r-p\r0% [1 InRelease gpgv 88.7 kB] [Connecting to archive.ubuntu.com (185.125.190.36\r                                                                               \rHit:2 https://cloud.r-project.org/bin/linux/ubuntu bionic-cran40/ InRelease\n",
            "\r0% [1 InRelease gpgv 88.7 kB] [Waiting for headers] [Waiting for headers] [Wait\r                                                                               \rIgn:3 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  InRelease\n",
            "\r0% [1 InRelease gpgv 88.7 kB] [Waiting for headers] [Waiting for headers] [Wait\r                                                                               \rHit:4 http://ppa.launchpad.net/c2d4u.team/c2d4u4.0+/ubuntu bionic InRelease\n",
            "\r0% [1 InRelease gpgv 88.7 kB] [Waiting for headers] [Waiting for headers] [Conn\r                                                                               \rHit:5 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu1804/x86_64  InRelease\n",
            "\r0% [1 InRelease gpgv 88.7 kB] [Waiting for headers] [Connecting to ppa.launchpa\r                                                                               \rHit:6 http://archive.ubuntu.com/ubuntu bionic InRelease\n",
            "Hit:7 https://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1804/x86_64  Release\n",
            "Hit:8 http://archive.ubuntu.com/ubuntu bionic-updates InRelease\n",
            "Hit:9 http://archive.ubuntu.com/ubuntu bionic-backports InRelease\n",
            "Hit:10 http://ppa.launchpad.net/cran/libgit2/ubuntu bionic InRelease\n",
            "Hit:11 http://ppa.launchpad.net/deadsnakes/ppa/ubuntu bionic InRelease\n",
            "Hit:12 http://ppa.launchpad.net/graphics-drivers/ppa/ubuntu bionic InRelease\n",
            "Reading package lists... Done\n",
            "Reading package lists... Done\n",
            "Building dependency tree       \n",
            "Reading state information... Done\n",
            "ncbi-blast+ is already the newest version (2.6.0-1).\n",
            "The following package was automatically installed and is no longer required:\n",
            "  libnvidia-common-460\n",
            "Use 'apt autoremove' to remove it.\n",
            "0 upgraded, 0 newly installed, 0 to remove and 28 not upgraded.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Белковая БД SwissProt:"
      ],
      "metadata": {
        "id": "xj26uQzvqBir"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz\n",
        "!gzip -d uniprot_sprot.fasta.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "dTY-RXPKp976",
        "outputId": "50c79388-3f26-4fd7-cd5e-c4868035fec8"
      },
      "execution_count": 33,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-21 13:34:19--  https://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz\n",
            "Resolving ftp.uniprot.org (ftp.uniprot.org)... 128.175.240.195\n",
            "Connecting to ftp.uniprot.org (ftp.uniprot.org)|128.175.240.195|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 91303010 (87M) [application/x-gzip]\n",
            "Saving to: ‘uniprot_sprot.fasta.gz’\n",
            "\n",
            "uniprot_sprot.fasta 100%[===================>]  87.07M   112MB/s    in 0.8s    \n",
            "\n",
            "2022-10-21 13:34:20 (112 MB/s) - ‘uniprot_sprot.fasta.gz’ saved [91303010/91303010]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "Пакет seqkit:"
      ],
      "metadata": {
        "id": "SN7fWAh6qVf1"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wget https://github.com/shenwei356/seqkit/releases/download/v2.1.0/seqkit_linux_amd64.tar.gz\n",
        "!tar -xzvf seqkit_linux_amd64.tar.gz\n",
        "!chmod a+x seqkit"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "KmWFwr01qK2n",
        "outputId": "736ac640-e5a1-4c77-968d-7d4cec9a32a0"
      },
      "execution_count": 34,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2022-10-21 13:34:24--  https://github.com/shenwei356/seqkit/releases/download/v2.1.0/seqkit_linux_amd64.tar.gz\n",
            "Resolving github.com (github.com)... 140.82.112.3\n",
            "Connecting to github.com (github.com)|140.82.112.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://objects.githubusercontent.com/github-production-release-asset-2e65be/52715040/ea4018ec-70e8-40ac-a4aa-3326b04e07c4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221021%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221021T133424Z&X-Amz-Expires=300&X-Amz-Signature=35a30bf1a6cfaf791cd0ede63b72a0f47beb129840cc0f4166a1aa4f9a80248b&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=52715040&response-content-disposition=attachment%3B%20filename%3Dseqkit_linux_amd64.tar.gz&response-content-type=application%2Foctet-stream [following]\n",
            "--2022-10-21 13:34:24--  https://objects.githubusercontent.com/github-production-release-asset-2e65be/52715040/ea4018ec-70e8-40ac-a4aa-3326b04e07c4?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20221021%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20221021T133424Z&X-Amz-Expires=300&X-Amz-Signature=35a30bf1a6cfaf791cd0ede63b72a0f47beb129840cc0f4166a1aa4f9a80248b&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=52715040&response-content-disposition=attachment%3B%20filename%3Dseqkit_linux_amd64.tar.gz&response-content-type=application%2Foctet-stream\n",
            "Resolving objects.githubusercontent.com (objects.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...\n",
            "Connecting to objects.githubusercontent.com (objects.githubusercontent.com)|185.199.108.133|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: 6650850 (6.3M) [application/octet-stream]\n",
            "Saving to: ‘seqkit_linux_amd64.tar.gz’\n",
            "\n",
            "seqkit_linux_amd64. 100%[===================>]   6.34M  --.-KB/s    in 0.08s   \n",
            "\n",
            "2022-10-21 13:34:24 (78.0 MB/s) - ‘seqkit_linux_amd64.tar.gz’ saved [6650850/6650850]\n",
            "\n",
            "seqkit\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Выполнение задания:**\n",
        "Предсказать расположение белок-кодирующих генов в геноме нашей бактерии\n",
        "* Предсказать белок-кодирующие гены в собранном геноме нашей бактерии с помощью ab initio программы GeneMarkS – комманда gms2.pl\n",
        "* По умолчанию координаты предсказанных генов сохраняются в файле gms2.lst\n",
        "* Обратите внимание, что GeneMarkS может выдавать последовательности генов (опция --fnn), а также автоматически их транслировать в белки (--faa)"
      ],
      "metadata": {
        "id": "WkUouEMyqkJf"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!./gms2_linux_64/gms2.pl  --seq Poil_gapClosed.fa  --genome-type bacteria  --fnn genes.fasta  --faa proteins.fasta"
      ],
      "metadata": {
        "id": "LiikYWSrqWVZ"
      },
      "execution_count": 41,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Выполнение задания**: Определяем функции белков нашей бактерии через сравнение с белками из T.oleivorans MIL-1\n",
        "* Сравнить все белки (протеом) нашей бактерии с аннотированными белками T. oleivorans:\n",
        "* Протеом T. oleivorans MIL-1 находится в файле T_oleivorans_MIL_1.proteins.fasta\n",
        "* Индексируем белки T.oleivorans MIL-1:"
      ],
      "metadata": {
        "id": "EeErKTFqwDAW"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!makeblastdb  -dbtype prot  -in T_oleivorans_MIL_1.proteins.fasta  -out T_oleivorans_MIL_1.proteins"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "yL5-z3JSvuK3",
        "outputId": "a7fed338-e0ee-4529-cd32-44104ddf0160"
      },
      "execution_count": 42,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "\n",
            "Building a new DB, current time: 10/21/2022 13:41:57\n",
            "New DB name:   /content/T_oleivorans_MIL_1.proteins\n",
            "New DB title:  T_oleivorans_MIL_1.proteins.fasta\n",
            "Sequence type: Protein\n",
            "Keep MBits: T\n",
            "Maximum file size: 1000000000B\n",
            "Adding sequences from FASTA; added 3666 sequences in 0.103845 seconds.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "* Определеям какие белки из нашей бактерии похожи на белки из T_oleivorans:"
      ],
      "metadata": {
        "id": "p-K5q931yABR"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!blastp  -query proteins.fasta  -db T_oleivorans_MIL_1.proteins  -evalue 1e-10  -outfmt 6  >  scaffolds.hits_from_MIL_1.txt"
      ],
      "metadata": {
        "id": "QYpGp3V_x1sj"
      },
      "execution_count": 43,
      "outputs": []
    },
    {
      "cell_type": "markdown",
      "source": [
        "**Выполнение задания**: Определяем функции белков нашей бактерии через поиск по БД SwissProt\n",
        "* Смотрим, какие белки нашей бактерии (из Баренцева моря) не имели похожих белков из бактерии T.oleivorans MIL-1\n",
        "* Для этого получаем список уникальных номеров генов бактерии из Барнцева моря, коротые имели хотя бы один похожий белок бактерии из Сицилли"
      ],
      "metadata": {
        "id": "Cxop4fO3yPib"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cut -f 1 scaffolds.hits_from_MIL_1.txt | sort -n | uniq > proteins.with_hits_from_MIL_1.txt\n",
        "!./seqkit grep  --invert-match  -f proteins.with_hits_from_MIL_1.txt  proteins.fasta   -o proteins.without_MIL_1.fasta"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "_L23pri0yDrM",
        "outputId": "079a2135-4867-4277-ed92-0391d3791c84"
      },
      "execution_count": 44,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[INFO]\u001b[0m 3339 patterns loaded from file\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "* Проверяем, что кол-ва белков сходятся\n",
        "* Общее кол-во белков в нашей бактерии:"
      ],
      "metadata": {
        "id": "0znfbI6fyjPV"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!grep '>' proteins.fasta | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "alFx803KyhMQ",
        "outputId": "15fefba2-3d2d-4251-8206-af1b5e006d39"
      },
      "execution_count": 45,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "3625\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\\> Кол-во белков, которые имели схожие белки из бактерии MIL-1"
      ],
      "metadata": {
        "id": "sPaTRp6_0Lu2"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!wc -l proteins.with_hits_from_MIL_1.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "rz5gnZdpyotF",
        "outputId": "f787a6bf-72e4-4061-c6b4-a58f78daf688"
      },
      "execution_count": 46,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "3339 proteins.with_hits_from_MIL_1.txt\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "\\> Какое кол-во белков остались без аннотации функции"
      ],
      "metadata": {
        "id": "BlevxRRZ0Vk_"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!grep '>' proteins.without_MIL_1.fasta | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "n6DAWAHs0UWb",
        "outputId": "5209e862-7e5a-4b69-cdb3-af10d56f2ad1"
      },
      "execution_count": 47,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "286\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "* Сравнить все белки (протеом) нашей бактерии с аннотированными белками из БД SwissProt\n",
        "* Индексируем белки SwissProt:"
      ],
      "metadata": {
        "id": "K7Wuou4g1pwZ"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!makeblastdb  -dbtype prot  -in uniprot_sprot.fasta  -out uniprot_sprot"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "PwkFknq81qN4",
        "outputId": "1a78fd50-12f8-41a8-b872-45cd6bef5d64"
      },
      "execution_count": 48,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "\n",
            "Building a new DB, current time: 10/21/2022 13:58:51\n",
            "New DB name:   /content/uniprot_sprot\n",
            "New DB title:  uniprot_sprot.fasta\n",
            "Sequence type: Protein\n",
            "Keep MBits: T\n",
            "Maximum file size: 1000000000B\n",
            "Adding sequences from FASTA; added 568363 sequences in 13.752 seconds.\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "* Определеям какие белки из нашей бактерии похожи на белки из БД SwissProt "
      ],
      "metadata": {
        "id": "kshvJ2I71z0q"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!time  blastp  -query proteins.without_MIL_1.fasta  -db uniprot_sprot  -evalue 1e-10  -outfmt 6  >  scaffolds.hits_from_SwissProt.txt"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "YGQ2KpoO1wvK",
        "outputId": "8bdafd17-0d42-4c93-c2ba-eef3dde54e66"
      },
      "execution_count": 50,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "real\t9m34.887s\n",
            "user\t9m33.117s\n",
            "sys\t0m0.512s\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "* Посмотрим, сколько белков удалось аннотировать через поиск по SwissProt:"
      ],
      "metadata": {
        "id": "qtm_mF6518Tj"
      }
    },
    {
      "cell_type": "code",
      "source": [
        "!cut -f 1 scaffolds.hits_from_SwissProt.txt | sort | uniq | wc -l"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "Aakd17sJ14ep",
        "outputId": "853c7561-71c9-44e7-bad9-b795f5fd851f"
      },
      "execution_count": 51,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "53\n"
          ]
        }
      ]
    },
    {
      "cell_type": "markdown",
      "source": [
        "* gms2.lst  --  координаты всех предсказанных генов\n",
        "* proteins.fasta  --  аминокислотные последовательности всех предсказанных генов\n",
        "* scaffolds.hits_from_MIL_1.txt  --  информация о схожести белков из нашей бактерии с белками из бактерии MIL-1\n",
        "* scaffolds.hits_from_SwissProt.txt  -- информация о схожести белков из нашей бактерии с белками из БД SwissProt"
      ],
      "metadata": {
        "id": "qyQsdbJR2C51"
      }
    }
  ]
}
